### ISSUE TITLE
Use socket API from portlib for JITaaS

### ISSUE THREAD

**Issue by mpirvu** [eclipse/openj9#6131]
Currently, the network communication layer in JITaaS is implemented using unix socket API. For portability we should replace that API with the one offered by the portlib and implemented in j9sock.c, unless it adds too much overhead (I doubt it)

**Comment by pdbain-ibm** [comment]
Currently, this is used only by the Harmony JDWP implementation (which I believe is now completely supplanted by the OpenJDK implementation). There is consideration now being given to removing this API.  I suggest discussing this with @pshipton.

**Comment by pshipton** [comment]
Indeed, we were hoping to remove the obsolete j9sock APIs from the OpenJ9 port library. As I understand it, at least some of the j9sock implementations are stale, which I think means not suitable for IPv6. A better approach is to implement a omrsock port library layer at the OMR project, containing what is needed to support JITaas. The parts of j9sock that are still relevant can be copied to omrsock.
